README.md

```diff
@@ -29,35 +29,110 @@ It must be configured before defining routes and other plugins in order to cover
   - `onResponse`
   - `onError`
 - Instruments automatically custom 404 Not Found handler
+- TODO: Clarify server shutdown/cleanup around otel metrics/traces
 
-Example:
+## Example
 
 ```js
-// ... in your OTEL setup
-const FastifyOtelInstrumentation = require('@fastify/otel');
+// otel.js
+// Sets up metrics, tracing, HTTP & Node runtime instrumentation, and host metrics.
+// Uses ConsoleSpanExporter for local debugging of spans.
+const { NodeSDK } = require('@opentelemetry/sdk-node')
+const { ConsoleSpanExporter } = require('@opentelemetry/sdk-trace-node')
+const { SimpleSpanProcessor } = require('@opentelemetry/sdk-trace-base')
+const { PrometheusExporter } = require('@opentelemetry/exporter-prometheus')
+const { HttpInstrumentation } = require('@opentelemetry/instrumentation-http')
+const { RuntimeNodeInstrumentation } = require('@opentelemetry/instrumentation-runtime-node')
+const { HostMetrics } = require('@opentelemetry/host-metrics')
+const { FastifyOtelInstrumentation } = require('@fastify/otel')
+const { metrics, trace } = require('@opentelemetry/api')
+const { resourceFromAttributes } = require('@opentelemetry/resources')
+const { SemanticResourceAttributes } = require('@opentelemetry/semantic-conventions')
+
+// Console exporter for development-time span inspection
+// We could use '@opentelemetry/exporter-trace-otlp-http' instead
+const traceExporter = new ConsoleSpanExporter()
+const spanProcessor = new SimpleSpanProcessor(traceExporter)
+
+const prometheusExporter = new PrometheusExporter({ port: 9091 })
+
+const sdk = new NodeSDK({
+  resource: resourceFromAttributes({
+    [SemanticResourceAttributes.SERVICE_NAME]: 'my-service-name',
+  }),
+  spanProcessor,
+  metricReader: prometheusExporter,
+  instrumentations: [
+    new HttpInstrumentation(),
+    new RuntimeNodeInstrumentation(),
+  ],
```

Comment:
It's a bit odd to me that `FastifyOtelInstrumentation` would not be used here under instrumentations instead, why is that the case?

Reply:
Only explanation I can find is https://github.com/fastify/fastify/issues/5862#issuecomment-2548188377

Reply:
Looks like you can (and should pass it here, fixed)